extensions/sql/contract-definition/store/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStore.java

```diff
@@ -0,0 +1,244 @@
+            try (Connection connection = dataSource.getConnection()) {
+                executeQuery(connection, query,
+                        definition.getId(),
+                        writeObject(definition.getAccessPolicy()),
+                        writeObject(definition.getContractPolicy()),
+                        writeObject(definition.getSelectorExpression()));
+            } catch (Exception exception) {
+                throw new EdcPersistenceException(exception);
+            }
+        });
+    }
+
+    @Override
+    public void update(ContractDefinition definition) {
+        Objects.requireNonNull(definition);
+
+        String query = String.format("UPDATE %s SET %s = ?, %s = ?, %s = ?, %s = ? WHERE contract_definition_id = ?",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ACCESS_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_CONTRACT_POLICY,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_SELECTOR);
+
+
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSource.getConnection()) {
+                executeQuery(connection, query,
+                        definition.getId(),
+                        writeObject(definition.getAccessPolicy()),
+                        writeObject(definition.getContractPolicy()),
+                        writeObject(definition.getSelectorExpression()),
+                        definition.getId());
+            } catch (Exception exception) {
+                throw new EdcPersistenceException(exception);
+            }
+        });
+    }
+
+    @Override
+    public void delete(String id) {
+        Objects.requireNonNull(id);
+
+        String query = String.format("DELETE FROM %s WHERE %s = ?",
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_TABLE,
+                SqlContractDefinitionTables.CONTRACT_DEFINITION_COLUMN_ID);
+
+        transactionContext.execute(() -> {
+            try (Connection connection = dataSource.getConnection()) {
+                executeQuery(connection, query,
+                        id);
+            } catch (Exception exception) {
+                throw new EdcPersistenceException(exception);
+            }
+        });
+    }
+
+    private String writeObject(Object object) {
+        try {
+            String className = object.getClass().getName();
+            String content = OBJECT_MAPPER.writeValueAsString(object);
+
+            Envelope envelope = new Envelope(className, content);
+
+            return OBJECT_MAPPER.writeValueAsString(envelope);
+        } catch (Exception e) {
+            throw new EdcException(e.getMessage(), e);
+        }
+    }
+
+    enum ContractDefinitionMapper implements ResultSetMapper<ContractDefinition> {
```

Comment:
This can be converted into a simple method and a method reference in line 67 and 88 which makes the code much cleaner.